keyboards/boston/keymaps/default/keymap.c

```diff
@@ -0,0 +1,223 @@
+/* Copyright 2020 Pylon
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+    /* Due to how the matrix is setup - the right split backspace key is, for firmware purposes the key to the right of up arrow, which on the actual keyboard is a blocker*/
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+   [0] = LAYOUT_all(
+	KC_MUTE, KC_F13 , KC_F14 , KC_F15 , KC_F16 , KC_F17 , KC_F18 , KC_F19 , KC_F20 , KC_F21 , KC_F22 , KC_F23 , KC_F24 ,                  KC_MPRV, KC_MPLY,  KC_MNXT ,          KC_INS , KC_HOME, KC_PGUP, 
+    KC_ESC , KC_F1  , KC_F2  , KC_F3  , KC_F4  , KC_F5  , KC_F6  , KC_F7  , KC_F8  , KC_F9  , KC_F10 , KC_F11 , KC_F12 ,                  KC_PSCR, KC_SLCK,  KC_PAUSE,          KC_DEL , KC_END , KC_PGDN,
+    KC_GRV , KC_1   , KC_2   , KC_3   , KC_4   , KC_5   , KC_6   , KC_7   , KC_8   , KC_9   , KC_0   , KC_MINS, KC_EQL ,  KC_BSPC, KC_BSPC,        KC_CALC,           KC_NLCK , KC_PSLS, KC_PAST, KC_PMNS,
+    KC_TAB , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   , KC_Y   , KC_U   , KC_I   , KC_O   , KC_P   , KC_LBRC, KC_RBRC,  KC_BSLS,                 RGB_TOG,           KC_P7   , KC_P8  , KC_P9  , KC_PEQL,
+    KC_CAPS, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   , KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN, KC_QUOT, KC_ENT ,  KC_ENT ,                 KC_MSEL,           KC_P4   , KC_P5  , KC_P6  , KC_PPLS,
+    KC_LSFT, KC_BSLS, KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   , KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_RSFT,                           KC_UP  ,           KC_P1   , KC_P2  , KC_P3  , KC_PENT,
+    KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC , KC_SPC , KC_SPC ,                   KC_RALT, KC_APP , KC_RCTL,                          KC_LEFT,  KC_DOWN, KC_RGHT,  KC_P0   , KC_P0  , KC_PDOT 
+    ),
+    
+   [1] = LAYOUT_all(
```

Comment:
All these empty layers are pretty pointless.

Reply:
Those are in there because I have Lighting Layers enabled for the RGB LED (with layer change via Fn key + knob), and I have colors coded in for specific layers, and want to provide this in the default keymap so that people who want this feature don't have to write all the Lighting Layers code.

Reply:
If I do not remove the empty layers will this PR not get approved?

Reply:
Probably this part of the PR checklist applies here:

> - default (and via) keymaps should be “pristine”
>     - bare minimum to be used as a “clean slate” for another user to develop their own user-specific keymap
>     - standard layouts preferred in these keymaps, if possible
> - submitters can have a personal (or bells-and-whistles) keymap showcasing capabilities in the same PR but it shouldn’t be embedded in the ‘default’ keymap

So you can add a separate keymap with all that custom code without cluttering the `default` keymap. Maybe even the `#define RGBLIGHT_LAYERS` part should also go into that keymap instead of being enabled at the keyboard level.

Also consider what the users of QMK Configurator would get — the configurator does not use the `default` keymap at all, it creates a new keymap with just the `keymaps` array initialized and nothing else.

Reply:
Thanks for the explanation - I stripped most of the empty layers out of the default keymap, and stripped most of the lighting layers code, and moved that to a separate RGB-Lighting-Layers keymap.